scripts/build/utils.js

```diff
@@ -0,0 +1,96 @@
+const path = require('path')
+const del = require('del')
+const { promisify } = require('util')
+const fs = require('fs')
+const makeRollupConfig = require('./rollup.config')
+const camelize = require('fbjs/lib/camelize')
+
+const readdir = promisify(fs.readdir)
+
+const rootPath = path.resolve(__dirname, '..', '..')
+
+exports.rootPath = rootPath
+
+exports.cleanDist = async function cleanDist(pkgPath) {
+  await del(`${pkgPath}/dist`)
+}
+
+exports.getPackages = async function getPackages() {
+  // we're intentionally not getting all the packages that are part of the monorepo
+  // we only want ones in packages
+  const packagePaths = (await readdir(path.join(rootPath, 'packages'))).map(
+    pkg => path.join(rootPath, 'packages', pkg)
+  )
+  const packages = packagePaths.map(packagePath => {
+    const fullPackagePath = path.resolve(rootPath, packagePath)
+    const ret = {
+      path: fullPackagePath,
+      pkg: require(path.resolve(fullPackagePath, 'package.json'))
+    }
+
+    ret.name = ret.pkg.name
+    ret.configs = []
+    if (ret.pkg.main && !ret.pkg.main.includes('src')) {
+      ret.configs.push({
+        config: makeRollupConfig(ret),
+        outputConfigs: getOutputConfigs(ret)
+      })
+    }
+    if (ret.pkg['umd:main']) {
+      ret.configs.push({
+        config: makeRollupConfig(ret, true, true),
```

Comment:
would prefer changing signature to:
```
makeRollupConfig(data, { isBrowser = false, isUmd = false } = {})
```

right now looking at this call site it's hard to tell what those booleans do

Reply:
not yet sure about this but - I can't find a package using "browser" field, so not sure how exactly we need to handle it, but a good alternative to transforming packages with babel and replacing stuff with browser values it might be clearer to introduce `BROWSER` env variable, produce code conditionally with `process.env` ternaries in the actual code and prepare 2 builds with appropriate `rollup-plugin-replace` setup